File: chap20/ZeroDUneFonction.js
// Ces programmes sont sous licence CeCILL-B V1.
var a, b, m,e;
e = 1E-5;
a = 2.0;
b = 4.0;
m = (a + b) / 2;
while (b - a > e & Math.abs(Math.sin(m)) > e) {
	if (Math.sin(a) * Math.sin(m) <= 0) {
		b = m;
	}   
	else { 
		a = m;
	}
	m = (a + b) / 2;
}
ISNprintln(m);

File: chap20/ZeroDUneFonction.py
# Ces programmes sont sous licence CeCILL-B V1.
from math import sin

seuil = 10**(-5)
inf = 2
sup = 4
milieu = (inf + sup) / 2
while sup - inf > seuil and abs(sin(milieu)) > seuil:
   if sin(inf) * sin(milieu) <= 0:
      sup = milieu
   else:
      inf = milieu
   milieu = (inf + sup) / 2
print(milieu)
